compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -432,6 +432,9 @@ pub fn impossible_predicates<'tcx>(
     debug!("impossible_predicates(predicates={:?})", predicates);
 
     let result = tcx.infer_ctxt().enter(|infcx| {
+        // Set tainted by errors to gracefully exit in case of overflow.
```

Comment:
I would mark this as HACK or something, since it's technically not using the flag correctly, right? I think it's fine otherwise.